BDKSwiftExampleWallet/Service/BDKSyncService/BDKSyncService.swift

```diff
@@ -0,0 +1,297 @@
+//
+//  BDKService2.swift
+//  BDKSwiftExampleWallet
+//
+//  Created by Rubens Machion on 16/05/25.
+//
+
+import BitcoinDevKit
+import Foundation
+
+typealias FullScanProgress = (UInt64) -> Void
+typealias SyncScanProgress = (UInt64, UInt64) -> Void
+
+enum SyncMode: String {
+    case esplora, kyoto
+}
+
+protocol BDKSyncService {
```

Comment:
would like to use a concrete type instead of protocol

similar to https://github.com/bitcoindevkit/BDKSwiftExampleWallet/blob/main/BDKSwiftExampleWallet/Service/BDK%20Service/BDKService.swift

happy to collaborate with you to do this, because I do think it would be nice to stay consistent with existing way the codebase tries to do this

reason for preference for concrete types over protocols comes a lot from pointfree folks (for example, see comment from Brandon https://github.com/pointfreeco/swift-composable-architecture/discussions/522)